configurate-core/src/main/java/ninja/leaping/configurate/transformation/TransformAction.java

```diff
@@ -37,6 +37,6 @@
      * @return A modified path, or null if the path is to stay the same
      */
     @Nullable
-    Object[] visitPath(@NonNull NodePath inputPath, @NonNull ConfigurationNode valueAtPath);
+    Object[] visitPath(ConfigurationTransformation.@NonNull NodePath inputPath, @NonNull ConfigurationNode valueAtPath);
```

Comment:
`@NonNull ConfigurationTransformation.NodePath`

Reply:
Not valid:

![](https://luck.sh/9bfkwm_pVg5EE-P85pMOYlmDlDI=)

The only other way to do it is to add an import statement for 
```java
import ninja.leaping.configurate.transformation.ConfigurationTransformation.NodePath
```

and then have

```java
Object[] visitPath(@NonNull NodePath inputPath, @NonNull ConfigurationNode valueAtPath);
```

but I thought that was unclear, as an interface with the same name already exists in the same package.

Reply:
To comment: the signature of this method will still have changed slightly (it was `SimpleConfigurationTransformation.NodePath`), but it actually points to the same class as `NodePath`'s outer class is `ConfigurationTransformation`, which is the superclass of `SimpleConfigurationTransformation`

I checked the bytecode from Nucleus which was compiled using configurate 3.2, it references `ConfigurationTransformation.NodePath` not `SimpleConfigurationTransformation.NodePath` - so this change should be fine.

Reply:
`ConfigurationTransformation. @NonNull NodePath`